platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -62,13 +62,13 @@ - (void)addToMapView:(MGLMapView *)mapView belowLayer:(MGLStyleLayer *)otherLaye
 
 - (void)removeFromMapView:(MGLMapView *)mapView
 {
-    _pendingLayer = nullptr;
-    self.rawLayer = nullptr;
-
     auto removedLayer = mapView.mbglMap->removeLayer(self.identifier.UTF8String);
     if (!removedLayer) {
         return;
     }
+    auto originalLayer = std::make_unique<mbgl::style::BackgroundLayer>(self.identifier.UTF8String);
+    _pendingLayer = std::move(originalLayer);
+    self.rawLayer = _pendingLayer.get();
```

Comment:
That is correct @1ec5. It is only to protect against that edge. I guess that could happen if you add a layer of type foo with identity bar and then (before adding) try to remove a layer of type baz with identity bar.